cmd/connector_celo.go

```diff
@@ -0,0 +1,101 @@
+//+build celo
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/celo-org/celo-blockchain/accounts/keystore"
+
+	commoncelo "github.com/keep-network/keep-common/pkg/chain/celo"
+	"github.com/keep-network/keep-common/pkg/chain/celo/celoutil"
+	"github.com/keep-network/keep-ecdsa/config"
+	eth "github.com/keep-network/keep-ecdsa/pkg/chain"
+	"github.com/keep-network/keep-ecdsa/pkg/chain/celo"
+)
+
+// Values related with balance monitoring.
+//
+// defaultBalanceAlertThreshold determines the alert threshold below which
+// the alert should be triggered.
+var defaultBalanceAlertThreshold = commoncelo.WrapWei(
+	big.NewInt(500000000000000000),
+)
+
+// defaultBalanceMonitoringTick determines how often the monitoring
+// check should be triggered.
+const defaultBalanceMonitoringTick = 10 * time.Minute
+
+func connectChain(
```

Comment:
All of this code, including defaults, can be hidden behind `ethereum.Connect` (see https://github.com/keep-network/keep-core/pull/2344#discussion_r582923438). This will let us minimize the code inside the build-tag-dependent connector and we will no longer have problems with `chain.Handle` interface and how to instantiate Ethereum-specific balance monitor.

Reply:
Since this is now a blocker for the chain refactor stuff, thoughts on merging this into master as-is (I'm making the assumption that it already works) and iterating on it on master?

Reply:
I haven't had a chance to review code generator changes today but I am fine with merging the rest of the code as-is and addressing comments in a follow-up PR. If you can look at the code generator changes @Shadowfiend and they look good to you, please go ahead and merge.